File: python/best_time_to_buy_and_sell.py
class Solution:
    def best_time_to_buy_and_sell(prices: list[int]) -> int:
        left = 0
        right = 1 
        current_profit, max_profit = 0, 0
        while right < len(prices):
            if prices[left] < prices[right]:
                current_profit = prices[right] - prices[left]
                if current_profit > max_profit:
                    max_profit = current_profit
            else:
                left = right
            right += 1
        return max_profit

        

    

File: Sliding Window/js/bestTimeToBuyAndSell.js
var maxProfit = function (prices) {
  let left = 0;
  let right = 1;
  let currentProfit = 0;
  let maxProfit = 0;

  while (right < prices.length) {
    if (prices[left] < prices[right]) {
      currentProfit = prices[right] - prices[left];
      if (currentProfit > maxProfit) {
        maxProfit = currentProfit;
      }
    } else {
      left = right;
    }
    right++;
  }
  return maxProfit;
};
